src/test/java/AccountTest.java

```diff
@@ -53,29 +53,44 @@ public void testAccountConstructorWithValidInput() {
     assertEquals(account.getAccountID(), ACCOUNT_ID);
     assertEquals(account.getVendorID(), VENDOR_ID);
     assertEquals(account.getEntity(), ENTITY);
-    assertEquals(account.getCurrency(), CURRENCY);   
-    assertEquals(account.getDirection(), DIRECTION);       
-    assertEquals(account.getLegacyAccountID(), LEGACY_ACCOUNT_ID);  
-    assertTrue("Account Constructor incorrectly set nextGenAccountID field.", 
-      account.getNextGenAccountID() == NEXT_GEN_ACCOUNT_ID);  
-    assertEquals(account.getMatchingMode(), MATCHING_MODE);  
-    assertEquals(account.getAggregationMode(), AGGREGATION_MODE);        
+    assertEquals(account.getCurrency(), CURRENCY);
+    assertEquals(account.getDirection(), DIRECTION);
+    assertEquals(account.getLegacyAccountID(), LEGACY_ACCOUNT_ID);
+    assertEquals("Account Constructor incorrectly set nextGenAccountID field.", account.getNextGenAccountID(), NEXT_GEN_ACCOUNT_ID);
+    assertEquals(account.getMatchingMode(), MATCHING_MODE);
+    assertEquals(account.getAggregationMode(), AGGREGATION_MODE);
   }
 
   /** Test that toJson returns a String representation of an Account object. */
   @Test
   public void testToJsonMethod() {
-    String expectedResponse = String.format("{\"legacy_account_id\":%s," +
-      "\"next_gen_customer_id\":%d,\"settlement_attributes\":{" +
-      "\"currency_code\":%s,\"direction\":%s,\"entity\":%s}," +
-      "\"settlement_config\":{\"matching_mode\":%s},\"account_id\":%s," +
-      "\"aggregation_mode\":%s}}", LEGACY_ACCOUNT_ID, NEXT_GEN_ACCOUNT_ID,
-      CURRENCY, DIRECTION, ENTITY, MATCHING_MODE, ACCOUNT_ID,AGGREGATION_MODE); 
+    String expectedResponse = String.format("{\"legacy_account_id\":\"%s\"," +
```

Comment:
It would be nice to validate that this string is valid JSON in this test. Consider adding that or at least a TODO

Reply:
(Same for test below)